### ISSUE TITLE
Support for collections with frozen types.

### ISSUE THREAD

**Issue by agarwal-nitesh** [scylladb/gocqlx#80]
`CREATE TYPE address(id text, street text, landmark text);`
`CREATE TABLE person(id text, name text, address list<frozen<address>>);`

```
type Person struct {
       Id string `db:"id"`
       Name string `db:"name"`,
       Address []Address `db:"address"`
}
type Address struct {
     Id string `db:"id"`,
     Street string `db:"street"`,
     Landmark string `db:"landmark"`
}
```

Insert fails for Address using BindStruct while BindMap is successful if the struct is converted to map[string]interface up to all nested levels.

**Comment by mmatczuk** [comment]
With gocql you must implement MarshalUDT and UnmarshalUDT functions to make it work see [1].
With gocqlx, however, you can do that in a very simple way

```go
// MarshalUDT implements UDTMarshaler.
func (u Address) MarshalUDT(name string, info gocql.TypeInfo) ([]byte, error) {
	f := gocqlx.DefaultMapper.FieldByName(reflect.ValueOf(u), name)
	return gocql.Marshal(info, f.Interface())
}

// UnmarshalUDT implements UDTUnmarshaler.
func (u *Address) UnmarshalUDT(name string, info gocql.TypeInfo, data []byte) error {
	f := gocqlx.DefaultMapper.FieldByName(reflect.ValueOf(u), name)
	return gocql.Unmarshal(info, data, f.Addr().Interface())
}
```

[1] https://github.com/gocql/gocql/blob/master/udt_test.go

**Comment by mmatczuk** [comment]
Fixed in #145
